hnn_core/optimization.py

```diff
@@ -351,24 +462,33 @@ def optimize_evoked(params, target_dpl, initial_dpl, maxiter=50,
     smooth_window_len : int
         The length of the hamming window (in samples) to smooth the
         simulated dipole waveform in each optimization step.
+    dt : float
+        The integration time step (ms) of h.CVode during simulation.
 
     Returns
     -------
-    params : dict
-        The optimized params dictionary.
+    net : Network instance
+        An instance of the Network object with the optimized configuration of
+        attached drives.
 
     Notes
     -----
     This optimization protocol utilizes the Constrained Optimization
     By Linear Approximation (COBYLA) method:
     https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.fmin_cobyla.html  # noqa
     """
-    from .parallel_backends import _BACKEND, JoblibBackend
 
-    if _BACKEND is None:
-        _BACKEND = JoblibBackend(n_jobs=1)
+    net = net.copy()
+
+    drive_names = [key for key in net.external_drives.keys()
+                   if net.external_drives[key]['type'] == 'evoked']
+    if len(drive_names) == 0:
+        raise ValueError('The current Network instance lacks any evoked '
```

Comment:
Is this tested?

Reply:
Good catch. Test added in https://github.com/jonescompneurolab/hnn-core/pull/446/commits/5bf14dc70e90910325cd039dec037f713e57d00d.